tst/EnergyPlus/unit/ReportCoilSelection.unit.cc

```diff
@@ -0,0 +1,365 @@
+	DataPlant::PlantLoop( 1 ).LoopSide( 1 ).TotalBranches = 1;
+	DataPlant::PlantLoop( 1 ).LoopSide( 1 ).Branch( 1 ).Comp.allocate( 1 );
+	DataPlant::PlantLoop( 1 ).LoopSide( 1 ).Branch( 1 ).TotalComponents = 1;
+	DataPlant::PlantLoop( 1 ).LoopSide( 2 ).Branch.allocate( 1 );
+	DataPlant::PlantLoop( 1 ).LoopSide( 2 ).TotalBranches = 1;
+	DataPlant::PlantLoop( 1 ).LoopSide( 2 ).Branch( 1 ).Comp.allocate( 1 );
+	DataPlant::PlantLoop( 1 ).LoopSide( 2 ).Branch( 1 ).TotalComponents = 1;
+
+	DataPlant::PlantLoop( 1 ).LoopSide( 1 ).Branch( 1 ).Comp( 1 ).NodeNumIn = 0;
+	DataPlant::PlantLoop( 1 ).LoopSide( 1 ).Branch( 1 ).Comp( 1 ).NodeNumOut = 0;
+	DataPlant::PlantLoop( 1 ).LoopSide( 2 ).Branch( 1 ).Comp( 1 ).NodeNumIn = chWInletNodeNum;
+	DataPlant::PlantLoop( 1 ).LoopSide( 2 ).Branch( 1 ).Comp( 1 ).NodeNumOut = chWOutletNodeNum;
+
+	//void
+	//finishCoilSummaryReportTable();
+
+	//void
+	//setCoilFinalSizes(  
+	//	std::string const & coilName, // user-defined name of the coil
+	//	std::string const & coilObjName , //  coil object name, e.g., Coil:Cooling:Water
+	//	Real64 const totGrossCap, // total capacity [W]
+	//	Real64 const sensGrossCap,// sensible capacity [W]
+	//	Real64 const airFlowRate, // design or reference or rated air flow rate [m3/s]
+	//	Real64 const waterFlowRate //design or reference or rated water flow rate [m3/s]
+	//);
+	//
+	//void
+	//setRatedCoilConditions(
+	//	std::string const & coilName,    // ! user-defined name of the coil
+	//	std::string const & coilObjName , //  coil object name, e.g., Coil:Cooling:Water
+	//	Real64 const RatedCoilTotCap,    // ! rated coil total capacity [W]
+	//	Real64 const RatedCoilSensCap,    // rated coil sensible capacity [W]
+	//	Real64 const RatedAirMassFlow,    // rated coil design air mass flow rate [m3/s]
+	//	Real64 const RatedCoilInDb,       // rated coil inlet air dry bulb at time of peak [C]
+	//	Real64 const RatedCoilInHumRat,   // rated coil inlet air humidity ratio [kgWater/kgDryAir]
+	//	Real64 const RatedCoilInWb,       // rated coil inlet air wet bulb [C]
+	//	Real64 const RatedCoilOutDb,      // rated coil outlet air dry bulb [C]
+	//	Real64 const RatedCoilOutHumRat,  // rated coil outlet air humidity ratio, [kgWater/kgDryAir]
+	//	Real64 const RatedCoilOutWb,      // rated coil outlet air wet bulb [C]
+
+	//	Real64 const RatedCoilOadbRef,    // rated DX coil outside air dry bulb reference [C]
+	//	Real64 const RatedCoilOawbRef,    // rated DX coil outside air wet bulb reference [C]
+	//	Real64 const RatedCoilBpFactor,   // rated coil bypass factor
+	//	Real64 const RatedCoilEff        // rated coil effectiveness
+	//);
+
+	Real64 airVdot ( 0.052 ); // air flow rate in m3/s
+	bool isAutoSized ( false ); // true if autosized
+	coilSelectionReportObj->setCoilAirFlow( coil1Name, coil1Type, airVdot, isAutoSized );
+	auto & c1 ( coilSelectionReportObj->coilSelectionDataObjs[ 0 ] );
+	EXPECT_EQ( coil1Name, c1->coilName_ );
+	EXPECT_EQ( coil1Type, c1->coilObjName );
```

Comment:
@Myoldmopar This works here and I can query c1 and see that's filled where it should be. But later on after coil 2 is created, c1 becomes undefined. What needs to be different here when declaring c1 to make it stick later?